PublicFolders/Migration/Sync-ModernMailPublicFolders.ps1

```diff
@@ -0,0 +1,846 @@
+                # Once the PF is MailEnabled and MailRecipientGuid of the first disconnected mepf found in the list is set, remaining are simply duplicate MEPFs
+                # Remove these duplicates and add email addresses to the mepf connected to it's PF
+                RemoveMEPFAndAddEmailAddresses $mailPublicFolder $publicFolder
+            }
+        }
+        Move-Item -Path $script:mailPublicFoldersDisconnectedFile -Destination $script:mailPublicFoldersDisconnectedFile.Replace('.txt', '_Processed.txt') -Force
+    }
+}
+
+function RemoveMEPFAndAddEmailAddresses() {
+    param ($duplicateMailPublicFolder, $publicFolder)
+
+    # Remove duplicate mepf
+    $folder = ([ADSI]("LDAP://$($duplicateMailPublicFolder.DistinguishedName)"))
+    $parent = ([ADSI]"$($folder.Parent)")
+    $parent.Children.Remove($folder)
+
+    # Add email addresses
+    $emailAddressesToAdd = @()
+    foreach ($emailAddress in $duplicateMailPublicFolder.EmailAddresses) {
+        if ($emailAddress.ToString().StartsWith("SMTP")) {
+            # Add the address as a secondary smtp address
+            $emailAddressesToAdd += $emailAddress.ToString().Substring($emailAddress.ToString().IndexOf(':') + 1)
+        } else {
+            $emailAddressesToAdd += $emailAddress.ToString()
+        }
+    }
+    Set-MailPublicFolder $publicFolder -EmailAddresses @{add =$emailAddressesToAdd }
+}
+
+################ DECLARING GLOBAL VARIABLES ################
+$script:isConnectedToExchangeOnline = $false
+$script:verbose = $VerbosePreference -eq [System.Management.Automation.ActionPreference]::Continue
+
+$script:csvSpecialChars = @("`r", "`n")
+$script:csvEscapeChar = '"'
+$script:csvFieldDelimiter = ','
+
+$script:ObjectsCreated = $script:ObjectsUpdated = $script:ObjectsDeleted = 0
+$script:NewSyncMailPublicFolderCommand = "New-RemoteSyncMailPublicFolder"
+$script:SetMailPublicFolderCommand = "Set-RemoteMailPublicFolder"
+$script:RemoveSyncMailPublicFolderCommand = "Remove-RemoteSyncMailPublicFolder"
+[char[]]$script:proxyAddressSeparators = ':', '@'
+$script:errorsEncountered = 0
+$script:authoritativeDomains = $null
+$script:mailEnabledSystemFolders = New-Object 'System.Collections.Generic.HashSet[Guid]'
+$script:WellKnownSystemFolders = @(
+    "\NON_IPM_SUBTREE\EFORMS REGISTRY",
+    "\NON_IPM_SUBTREE\OFFLINE ADDRESS BOOK",
+    "\NON_IPM_SUBTREE\SCHEDULE+ FREE BUSY",
+    "\NON_IPM_SUBTREE\schema-root",
+    "\NON_IPM_SUBTREE\Events Root")
+$script:foldersToMailDisableFile = Join-Path $PWD "FoldersToMailDisable.txt"
+$script:mailPublicFolderOrphansFile = Join-Path $PWD "MailPublicFolderOrphans.txt"
+$script:mailPublicFolderDuplicatesFile = Join-Path $PWD "MailPublicFolderDuplicates.txt"
+$script:emailAddressMergeScriptFile = Join-Path $PWD "AddAddressesFromDuplicates.ps1"
+$script:mailDisabledWithProxyGuidFile = Join-Path $PWD "MailDisabledWithProxyGuid.txt"
+$script:mailPublicFoldersDisconnectedFile = Join-Path $PWD "MailPublicFoldersDisconnected.txt"
```

Comment:
Is there a better way to keep these in sync between the two scripts? Example, in `ValidateMailEnabledPublicFolders.ps1` you create a script variable that can be used here for all the exported problem data. This way you don't forget to keep them in sync, unless there isn't really another scenario that we think can occur moving forward.

Reply:
I didn't write this script and didn't even know it depended on those files from my own script until I started migrating these. I don't plan to do major refactoring unless we start getting escalations on the sync script.